### PR TITLE
Use Delta's SchemaUtils.reportDifferences to report schema differences

### PR DESCRIPTION
**Pull request by zsxwing**
This is one example of the new error message:

```
MetaException(message:The Delta table schema is not the same as the Hive schema:

Specified schema is missing field(s): a
Specified schema has additional field(s): e

Delta table schema:
root
-- a: integer (nullable = true)
-- b: string (nullable = true)
-- c: string (nullable = true)

Hive schema:
e: int
c: string
b: string

Please update your Hive table's schema to match the Delta table schema.)
```

### CODE REVIEW

File: hive/src/main/scala/org/apache/spark/sql/delta/DeltaHelper.scala [hunk 1]
```diff
@@ -173,64 +175,78 @@ object DeltaHelper extends Logging {
    */
   @throws(classOf[MetaException])
   def checkTableSchema(deltaSchema: StructType, hiveSchema: StructTypeInfo): Unit = {
-    // TODO How to check column nullables?
-    if (!isSameStructType(deltaSchema, hiveSchema)) {
-      throw metaInconsistencyException(deltaSchema, hiveSchema)
+    val deltaType = normalizeSparkType(deltaSchema).asInstanceOf[StructType]
+    val hiveType = hiveTypeToSparkType(hiveSchema).asInstanceOf[StructType]
+    if (deltaType != hiveType) {
```
Comment (windpiger) [inline hive/src/main/scala/org/apache/spark/sql/delta/DeltaHelper.scala]: what about if the order fields of the hiveschema&deltaschema are difference?

[the remark this answers is not in the document]
Comment (zsxwing) [reply hive/src/main/scala/org/apache/spark/sql/delta/DeltaHelper.scala]: `StructType`'s `equals` will return false if the order is not the same.

[the remark this answers is not in the document]
Comment (zsxwing) [reply hive/src/main/scala/org/apache/spark/sql/delta/DeltaHelper.scala]: https://github.com/delta-io/connectors/blob/master/hive/src/test/scala/io/delta/hive/HiveConnectorSuite.scala#L165 we already have a test for this case.

[the remark this answers is not in the document]
Comment (windpiger) [reply hive/src/main/scala/org/apache/spark/sql/delta/DeltaHelper.scala]: I think it is no need to check the field order , we can just check filed name&type

[the remark this answers is not in the document]
Comment (zsxwing) [reply hive/src/main/scala/org/apache/spark/sql/delta/DeltaHelper.scala]: Yep. Good point. However, when we calculate the partition column location, we assume the column order is the same implicitly. I created #25 to improve it.

**Comment by windpiger** [comment]
LGTM

**Pull request merged**